UnitTests/TextMaskFieldTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Xunit;
+
+namespace Terminal.Gui {
+	public class TextMaskFieldTests {
+		private TextMaskField _textMaskField_Mask;
+		private TextMaskField _textMaskField_Regex;
+
+		public TextMaskFieldTests ()
+		{
+			Application.Init (new FakeDriver (), new FakeMainLoop (() => FakeConsole.ReadKey (true)));
+
+			_textMaskField_Mask = new TextMaskField ("+(999) 000-0000");
+			_textMaskField_Regex = new TextMaskField ("^[A-Z][0-9]$", TextMaskField.TextMaskType.Regex);
+		}
+
+		[Fact]
+		public void Validate_Regex_Field_RegexValidateInput_False ()
+		{
+			// Allow input. Validates field.
+			_textMaskField_Regex.RegexValidateInput = false;
+
+			_textMaskField_Regex.ProcessKey (new KeyEvent ((Key)(int)'E', new KeyModifiers ()));
+			Assert.Equal ("E", _textMaskField_Regex.Text);
+			Assert.False (_textMaskField_Regex.IsValid);
```

Comment:
Seems to me a few more tests to exhaustively prove you are filtering bad (dangerous?) regex might be in order.